generative/maisi/scripts/augmentation.py

```diff
@@ -0,0 +1,555 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Sequence
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+from monai.transforms import Rand3DElastic, RandAffine, RandZoom
+from monai.utils import ensure_tuple_rep
+from torch import Tensor
+
+from .utils import dilate_one_img, erode_one_img
+
+MAX_COUNT = 100
+
+
+def initialize_tumor_mask(volume: Tensor, tumor_label: Sequence[int]) -> Tensor:
+    """
+    Initialize tumor mask for tumor augmentation.
+
+    Args:
+        volume: input 3D multi-label mask, [1,H,W,D] torch tensor.
+        tumor_label: tumor label in whole_mask, list of int.
+
+    Return:
+        tumor_mask_, initialized tumor mask, [1,H,W,D] torch tensor.
+    """
+    tumor_mask_ = torch.zeros_like(volume, dtype=torch.uint8)
+    for idx, label in enumerate(tumor_label):
+        tumor_mask_[volume == label] = idx + 1
+    return tumor_mask_
+
+
+def finalize_tumor_mask(distorted_mask: Tensor, organ_mask: Tensor, threshold_tumor_size: float):
+    """
+    Try to generate the final tumor mask by combining the augmentd tumor mask and organ mask.
+    Need to make sure tumor is inside of organ and is larger than threshold_tumor_size.
+
+    Args:
+        distorted_mask: input 3D binary tumor mask, [1,H,W,D] torch tensor.
+        organ_mask: input 3D binary organ mask, [1,H,W,D] torch tensor.
+        threshold_tumor_size: threshold tumor size, float
+
+    Return:
+        tumor_mask, [H,W,D] torch tensor; or False if the size did not qualify
```

Comment:
It returns `None` instead of `False` in my code reading

Reply:
Corrected. Thanks